src/Components/Modals/DeleteAdditionalRepositoriesModal.js

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import {
+  Button,
+  Modal,
+  ModalVariant,
+  TextContent,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons';
+import propTypes from 'prop-types';
+import useNotifications from '../../hooks/useNotifications';
+import { useQueryClient } from 'react-query';
+import useDeleteAdditionalRepositories from '../../hooks/useDeleteAdditionalRepositories';
+
+const DeleteAdditionalRepositoriesModal = (props) => {
+  const {
+    isOpen,
+    handleModalToggle,
+    name,
+    repositoryNameToDelete,
+    repositoryLabelToDelete,
+    setDeletedRepositories,
+  } = props;
+  const { addSuccessNotification, addErrorNotification } = useNotifications();
+  const { mutate, isLoading } = useDeleteAdditionalRepositories();
+  const queryClient = useQueryClient();
+
+  const deleteAdditionalRepositories = (
+    name,
+    repositoryNameToDelete,
+    repositoryLabelToDelete
+  ) => {
+    console.log(name, repositoryNameToDelete, repositoryLabelToDelete);
+    const payload = [
+      {
+        repositoryLabel: repositoryLabelToDelete,
+        repositoryName: repositoryNameToDelete,
+      },
+    ];
+    console.log('Request Payload:', payload);
+
+    mutate(
+      { name, payload },
+      {
+        onSuccess: (data, queryName) => {
+          const updatedData = data?.filter(
+            (entry) => entry.repositoryName !== repositoryNameToDelete
+          );
+          queryClient.setQueryData(queryName, updatedData);
+          addSuccessNotification(
+            `Additional repository ${repositoryNameToDelete} deleted`
+          );
+          handleModalToggle(true);
+          setDeletedRepositories((prevState) => [
+            ...prevState,
+            {
+              repositoryName: repositoryNameToDelete,
+              repositoryLabel: repositoryLabelToDelete,
+            },
+          ]);
```

Comment:
If we get rid of the filtering I commented on then this is also no longer necessary